modules/mono/csharp_script.cpp

```diff
@@ -119,6 +119,8 @@ void CSharpLanguage::init() {
 #ifdef TOOLS_ENABLED
 	GLOBAL_DEF("dotnet/project/solution_directory", "");
 	GLOBAL_DEF(PropertyInfo(Variant::INT, "dotnet/project/assembly_reload_attempts", PROPERTY_HINT_RANGE, "1,16,1,or_greater"), 3);
+	GLOBAL_DEF(PropertyInfo(Variant::PACKED_STRING_ARRAY, "dotnet/build/custom_compiler_arguments"), PackedStringArray());
+	GLOBAL_DEF(PropertyInfo(Variant::PACKED_STRING_ARRAY, "dotnet/build/custom_publish_arguments"), PackedStringArray());
```

Comment:
A similar setting (`dotnet/editor/custom_exec_path_arg`) is defined as a [`String`](https://github.com/godotengine/godot/blob/46c495ca21f40f57a7fb9c7cde6143738f1652d4/modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs#L609-L614) instead of a `PackedStringArray`. I don't know which one would be better, but maybe we want to be consistent.

Also, as mentioned by @lfshr, the `dotnet/build/custom_publish_arguments` setting would be better as an **export option** rather than a **project setting**, since that would allow you to specify a different value for each export preset.